TrafficSimulator/Assets/RoadGenerator/Core/Runtime/Objects/Intersection.cs

```diff
@@ -102,6 +121,116 @@ private List<RoadNode> GetJunctionNodes(Road road)
             return junctionNodes;
         }
 
+        // Creates a gameobject container for the intersection to keep track of all assets in the intersection
+        private void CreateIntersectionContainer()
+        {
+            FlowContainer = new GameObject("FlowContainer");
+            FlowContainer.transform.parent = IntersectionObject.transform;
+            FlowContainer.transform.localPosition = Vector3.zero;
+            FlowContainer.transform.localRotation = Quaternion.identity;
+            FlowContainer.transform.localScale = Vector3.one;
+
+            CreateTrafficLightController();
+        }
+
+        // Creates a gameobject container to control the intersections traffic lights
+        private void CreateTrafficLightController()
+        {
+            TrafficLightController = Instantiate(RoadSystem._defaultTrafficLightControllerPrefab, Vector3.zero, Quaternion.identity);
+            TrafficLightController.transform.parent = FlowContainer.transform;
+            TrafficLightController.transform.localScale = Vector3.one;
+        }
+
+        private TrafficLight SpawnTrafficLight(Vector3 position, Quaternion rotation)
+        {
+            GameObject trafficLight = Instantiate(RoadSystem._defaultTrafficLightPrefab, position, rotation);
+            trafficLight.transform.parent = TrafficLightController.transform;
+            return trafficLight.GetComponent<TrafficLight>();
+        }
+
+        // Finds intersections junction nodes and assigns traffic lights to them
+        private void AssignTrafficLights()
+        {
+            List<RoadNode> road1IntersectionNodes = new List<RoadNode>();
+            List<RoadNode> road2IntersectionNodes = new List<RoadNode>();
+
+            RoadNode road1Node = Road1.StartNode;
+            RoadNode road2Node = Road2.StartNode;
+
+            //Find the edge nodes of the roads
+            while (road1Node != null)
+            {
+                if (road1Node.Type == RoadNodeType.JunctionEdge && (road1Node.Position == Road1AnchorPoint1 || road1Node.Position == Road1AnchorPoint2))
+                {
+                    road1IntersectionNodes.Add(road1Node);
+                }
+                road1Node = road1Node.Next;
+            }
+            while (road2Node != null)
+            {
+                if (road2Node.Type == RoadNodeType.JunctionEdge && (road2Node.Position == Road2AnchorPoint1 || road2Node.Position == Road2AnchorPoint2))
+                {
+                    road2IntersectionNodes.Add(road2Node);
+                }
+                road2Node = road2Node.Next;
+            }
+
+            List<RoadNode> longRoad = new List<RoadNode>();
```

Comment:
hard to understand name. The length of the road have nothing to do with the variable.